### PR TITLE
Pass operator as a runtime parameter to _d_assert_fail

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
The operator is not required at compile time and can hence be passed as a runtime parameter to create less unique template instations.

This commit moves the operator from the first template parameters to the first runtime parameter (hence several index changes).

---

~~Depends on dlang/druntime#3319~~

### CODE REVIEW

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -6165,24 +6165,26 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
...
+                            (*es)[1] = maybePromoteToTmp(ne.e1);
                             break;
                         }
                     }
                 }
                 else
-                {   // Simply format exp.e1
-                    (*es)[0] = maybePromoteToTmp(exp.e1);
+                {
                     // No special treatment for other operators to avoid redundant template instantions
```
Comment (Geod24) [inline src/dmd/expressionsem.d:6176]: This comment is not accurate anymore

**Review by Geod24 — APPROVED** [review]
LGTM, just one minor nitpick

File: src/dmd/expressionsem.d (hunk 1 above)
Comment (MoonlightSentinel) [reply src/dmd/expressionsem.d:6176]: Fixed

**Review by PetarKirov — APPROVED** [review]
LGTM too

**Pull request merged**